### DISCUSSION TITLE
微信公众号辅助增强

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#5851] [插件说明]
描述
---

优化微信公众号在简悦阅读模式下的显示效果，包括：适配小红书类页面布局、保留文字颜色、显示公式、去除多余空格等。

地址
---

[https://simpread.ksria.cn/plugins/details/N3EznnUXeV](https://simpread.ksria.cn/plugins/details/N3EznnUXeV)

截图
---

![image](https://github.com/Kenshin/simpread/assets/81074/e84b908f-e038-4dda-a921-17c22614c411)

视频
---

https://github.com/Kenshin/simpread/assets/81074/9ffd97b4-1cb4-428b-a883-621d94c1f6c1

选项
---

随便进入一个页面 e.g. https://sspai.com/post/71576 并进入 **阅读模式 → 右下角 → 动作 → 插件触发器**

按照下图所示即可查看此插件的选项页。（注意：如果你的界面没有插件触发器，请根据此 [解决方案](https://github.com/Kenshin/simpread/discussions/2342) 修复。

<img width="1106" alt="image" src="https://github.com/user-attachments/assets/d5edd753-78be-40e3-92d7-5e4ad828c47a">

优化目的
---

为了更易于阅读以及导出（到其它生产力工具）。

优化内容
---

- 完美识别公式

- 大片空白的问题

- 个别图片变得非常巨大的问题

- 个别页面无法成功获取正文的问题

- 小红书样式无法使用阅读模式的问题

- 修复公众号公式与代码段无法正常显示的问题

- 去除多余空白行的问题

- 去除不易于导出的内容，如：视频、音频、小程序、公众号推荐等。

功能
---

- [优化类似小红书布局的页面](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6859735)（默认开启）

- 显示原图（默认情况下公众号的图片不是原图，开启后可显示为原图。

- [强制模式](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6736529)（如果经常查看一些界面比较花哨的公众号，请开启此选项）

- 图片居中（开启 **强制模式** 后生效，如果不开启强制模式的话，原本图片就是居中显示）

- [显示公众号作者等信息](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6849468)
   
- [去除多余空白行](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6850107)（默认关闭）

- [识别翻页/目录](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-10914017)

- [手动优化模式](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6850622)

   > 强制模式关闭后，可手动启用，启用后即便以下选项关闭状态也会自动优化

   <img width="461" alt="image" src="https://github.com/user-attachments/assets/d2157afb-4b2f-4264-98d4-bb0360c7436f">

- [排除列表](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-10037129)

安装与升级
---

- 没有【使用同步助手 · 自动同步】的话，可通过 [此链接](https://simpread.ksria.cn/plugins/details/N3EznnUXeV) 安装。

- 已经【使用同步助手 · 自动同步】的话，请使用 [插件管理器](https://www.yuque.com/kenshin/simpread/fcw60u#ymS1C) 安装。

更新日志
---

### 2.0.1

- `Add` 一些新的干扰元素

- `Add` 一些新的多余空格的元素

### 2.0.0

- `Add` 显示原图

- `Add` 图片居中显示

- `Add` 可识别 翻页与目录

- `Optimize` 当使用强制模式时，图片居中设置

- `Optimize` 小红书布局方案，使其更易于阅读

- `Optimize` 去除不易于导出的内容，如：视频、音频、小程序、公众号推荐等

- `Optimize` 优化选项卡 UI 到 2.0 版

### 1.1.0

- `Add` [排除列表](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-10037129)

### 1.0.1

- `Optimize` 代码段显示错误的问题

### 1.0.0

- [优化类似小红书布局的页面](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6859735)

- [强制模式](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6736529)（如果经常查看一些界面比较花哨的公众号，请开启此选项）

- [显示公众号作者等信息](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6849468)

- [去除多余空白行](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6850107)

- [手动优化模式](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6850622)

**Comment by Kenshin** [comment]
# 强制模式

如果经常查看一些界面比较花哨的公众号，请开启此选项。

## 为什么要有此功能

某些公众号为了让版面「更好看」会加入很多乱七八糟的内容，当进入阅读模式后简悦会自动剔除它们，但剔除后在某些公众号上反而效果不好，所以增加了此选项。

开启后，会将公众号的页面直接显示在阅读模式上，实现了完全还原页面效果。

## 注意

在页面上可以实现还原，但导入到 Obsidian、Notion 等生产力工具后也会把这些内容一并导入，所以有导入诉求的话，不建议开启此选项。

## 界面布局

严格按照原公众号界面显示

<img width="1497" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/57e02a0f-ab43-41b1-8664-89ab2e06ba13">

## 公式

下面是进入阅读模式后可正常显示公式（注意：虽然在阅读模式下可正常显示公式，但如果导出到 Obsidian 还是会有问题，因为 Markdown 的限制导致，这时使用导出 HTML（或离线 HTML）

![image](https://github.com/Kenshin/simpread/assets/81074/8b6085d6-d66f-4352-b05d-cbc809a76c3a)

## 代码段

![image](https://github.com/Kenshin/simpread/assets/81074/ee6f1512-63ed-442e-b638-91db35d3d463)

**Comment by Kenshin** [comment]
# 显示公众号作者等信息

开启后的效果，此结构可正常导入到任意生产力工具 e.g. Obsidian、Notion 等。

<img width="731" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/ecd273ee-e517-4c75-8ae6-335d83ffac23">

**Comment by Kenshin** [comment]
# 去除多余空白行

此选项默认为关闭，如有需要请开启。

## 效果

<img width="1271" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/a1a15e5b-9fa4-4007-bda4-480fbb803497">

**Comment by Kenshin** [comment]
# 手动优化模式

并不是每个公众号都需要优化，如果你不经常阅读公众号或者你的公众号在阅读模式下没问题，只是偶尔有效果不好的问题，可关闭这些选项，在有需要时选择手动启用。

<img width="466" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/e0721d7f-400a-4c2b-96f2-f97c312edc9a">

## 如何启用

在有需要优化的地方，右下角 → 动作 → 插件触发器 → 手动优化微信公众号

<img width="450" alt="image" src="https://github.com/Kenshin/simpread/assets/81074/7d0d71dd-5049-4806-81e7-47d09e4f8571">

## 效果

https://github.com/Kenshin/simpread/assets/81074/1ffc07c4-307f-4529-970b-75699e7e6bc8

**Reply by here-ted** [reply]
"手动优化公众号"能否增加快捷键呢

**Reply by Kenshin** [reply]
@here-ted 

暂时没有，不过你可以试试这个插件 https://github.com/Kenshin/simpread/discussions/1600

PS：后续我会考虑升级这个插件，使其可以支持任意快捷键的功能。

**Comment by Kenshin** [comment]
# 优化类似小红书的布局

可以将以图为主的公众号生成阅读模式，如下图所示。

![image](https://github.com/Kenshin/simpread/assets/81074/5dfef6cc-f9da-4ae2-98bf-bd0c9844ff87)

**Comment by Kenshin** [comment]
# 排除列表

<img width="469" alt="image" src="https://github.com/user-attachments/assets/767470dd-813e-4a17-9997-76c74914dd03">

## 描述

仅针对 [强制模式](https://github.com/Kenshin/simpread/discussions/5851#discussioncomment-6736529)，如果某些页面使用强制模式效果反而不佳，请加黑名单，加入后不会执行 **强制模式** 的优化方式，其它优化设置不受影响。

## 特别说明

即便加入了排除列表，但仍可使用 **手动模式** 方式优化。（位置：右下角 → 动作 → 插件触发器）

<img width="244" alt="image" src="https://github.com/user-attachments/assets/d6f6d82d-c213-47b0-9b01-7ad9d9fe1515">

## 添加方式

有两个位置可以添加：

1. 右下角 → 动作 → 插件触发器 → 微信公众号辅助增强选项

    <img width="469" alt="image" src="https://github.com/user-attachments/assets/767470dd-813e-4a17-9997-76c74914dd03">

2. 右下角 → 动作 → 插件触发器 → 将当前页面加入排除列表

    <img width="287" alt="image" src="https://github.com/user-attachments/assets/18d237ec-4511-4a3e-9006-c97eb5159e5e">

**Comment by Kenshin** [comment]
# 识别翻页/目录

如果你经常阅读的公众号具有目录（以及上下篇）的功能，可以开启这个功能，在阅读模式下就可以实现翻页。

此选项默认关闭，请手动开启

<img width="466" alt="image" src="https://github.com/user-attachments/assets/273d1c54-806b-49d2-b3f6-bda27efcf8bf">

## 识别前（公众号原页面效果）

<img width="723" alt="image" src="https://github.com/user-attachments/assets/c9e727f6-e2a4-4de0-a05d-6cc6d0e9be7d">

## 识别后（进入阅读模式）

下图除了可以正常识别翻页外，还可以点击进行翻页操作。

<img width="1231" alt="image" src="https://github.com/user-attachments/assets/4dd64c99-b937-4e1f-b289-6da72f2ea658">
